From 7517c89360d2b0f6cd8f87aeffc8f4e6528eba3f Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Thu, 15 Oct 2020 11:39:19 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3568 add io-domain for evb

The io-domain node should be matched to hardware power tree, lookup to
IO Power Domain Map on EVB Schematics For RK3568 file.

Change-Id: I03a0a66b3b21de055912624b893322fa4880a262
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 .../boot/dts/rockchip/rk3568-android.dtsi     | 16 ++++
 arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi  | 90 ++++++++++++++++++-
 .../arm64/boot/dts/rockchip/rk3568-linux.dtsi | 16 ++++
 3 files changed, 121 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
index 18c97bd604e4..16c876632278 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
@@ -22,3 +22,19 @@
 		status = "okay";
 	};
 };
+
+&io_domains {
+	status = "okay";
+	vccio1-supply = <&vccio1>;
+	vccio3-supply = <&vccio3>;
+	vccio4-supply = <&vccio4>;
+	vccio5-supply = <&vccio5>;
+	vccio6-supply = <&vccio6>;
+	vccio7-supply = <&vccio7>;
+};
+
+&pmu_io_domains {
+	status = "okay";
+	pmuio1-supply = <&vccpmuio1>;
+	pmuio2-supply = <&vccpmuio2>;
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
index 1f6f2b88d5ad..f536fdcdc18b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
@@ -37,6 +37,94 @@
 		regulator-max-microvolt = <5000000>;
 		vin-supply = <&dc_12v>;
 	};
+
+	/*
+	 * Configure for io-domain node according to Power Sequence/IO Domain Map
+	 * For example, RK3568 EVB V10
+	 * -vccpmuio1,vccpmuio2
+	 * -vccio1,vccio2,vccio3,vccio4,vccio5,vccio6,vccio7
+	 */
+	vccpmuio1: vcc-pmuio1 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccpmuio1";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vcc3v3_pmu>;
+	};
+
+	vccpmuio2: vcc-pmuio2 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccpmuio2";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vcc3v3_pmu>;
+	};
+
+	vccio1: vccio1 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio1";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vccio_acodec>;
+	};
+
+	/* vccio2 reserved, selected by pin "FLASH_VOL_SEL" */
+
+	vccio3: vccio3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio3";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vccio_sd>;
+	};
+
+	vccio4: vccio4 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio4";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		vin-supply = <&vcc_1v8>;
+	};
+
+	vccio5: vccio5 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio5";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vcc_3v3>;
+	};
+
+	vccio6: vccio6 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio6";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		vin-supply = <&vcc_1v8>;
+	};
+
+	vccio7: vccio7 {
+		compatible = "regulator-fixed";
+		regulator-name = "vccio7";
+		regulator-always-on;
+		regulator-boot-on;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vcc_3v3>;
+	};
 };
 
 &i2c0 {
@@ -212,7 +300,7 @@
 			vccio_sd: LDO_REG5 {
 				regulator-always-on;
 				regulator-boot-on;
-				regulator-min-microvolt = <3300000>;
+				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vccio_sd";
 				regulator-state-mem {
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
index 18c97bd604e4..16c876632278 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
@@ -22,3 +22,19 @@
 		status = "okay";
 	};
 };
+
+&io_domains {
+	status = "okay";
+	vccio1-supply = <&vccio1>;
+	vccio3-supply = <&vccio3>;
+	vccio4-supply = <&vccio4>;
+	vccio5-supply = <&vccio5>;
+	vccio6-supply = <&vccio6>;
+	vccio7-supply = <&vccio7>;
+};
+
+&pmu_io_domains {
+	status = "okay";
+	pmuio1-supply = <&vccpmuio1>;
+	pmuio2-supply = <&vccpmuio2>;
+};
-- 
2.35.3

